rust/noosphere-cli/tests/gateway.rs

```diff
@@ -33,11 +32,92 @@ use noosphere_cli::native::{
     },
     workspace::Workspace,
 };
+use noosphere_core::{
+    authority::{generate_ed25519_key, Authorization, SUPPORTED_KEYS},
+    data::{ContentType, Did, MemoIpld},
+    view::{Sphere, SphereMutation},
+};
+use noosphere_ns::{DHTKeyMaterial, Multiaddr, NSRecord, NameSystem, NameSystemBuilder};
+use noosphere_storage::{BlockStore, MemoryStorage, SphereDb, Storage};
+
+async fn create_bootstrap_node() -> Result<(
+    Ed25519KeyMaterial,
+    SphereDb<MemoryStorage>,
+    NameSystem<MemoryStorage, Ed25519KeyMaterial>,
+    Vec<Multiaddr>,
+)> {
+    let db = SphereDb::new(&MemoryStorage::default()).await?;
+    let key = generate_ed25519_key();
+    let mut node: NameSystem<MemoryStorage, Ed25519KeyMaterial> = NameSystemBuilder::default()
+        .key_material(&key)
+        .store(&db)
+        .listening_port(thread_rng().gen_range(49152..65525))
+        .peer_dialing_interval(1)
+        .build()?;
+    node.connect().await?;
+    let address = node.p2p_address().unwrap().to_owned();
+    Ok((key, db, node, vec![address]))
+}
+
+/// Copies the authorization UCAN associated with `context` into the `dest_store`.
+/// Sort of a hack to propagate the CID from local sphere to the name system's
+/// store in order to validate.
+async fn copy_authorization_ucan_token<K: DHTKeyMaterial + 'static, S1: Storage, S2: Storage>(
```

Comment:
Unsure of the best way to propagate the delegation CID to bootstrap NS. If PUBLISH called after a PUSH, the delegation should be available on IPFS, though the NS doesn't have a running KuboClient or anything yet. So still a few things need to be thought out w/r/t validating things in the NS relying on IPFS..